internal/terraform/node_resource_plan_instance.go

```diff
@@ -287,7 +298,11 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 		}
 
 		deferrals := ctx.Deferrals()
-		if !deferrals.ShouldDeferResourceInstanceChanges(n.Addr) {
+
+		if deferrals.IsResourceInstanceDeferred(n.Addr) {
+			// This resource instance is already deferred, so we don't need to
+			// do anything else with it here.
```

Comment:
nit: maybe extend this comment to say something like "This resource instance is already deferred, probably because it was deferred during the import any/or refresh state"?